python_tsp/exact/branch_and_bound/node.py

```diff
@@ -0,0 +1,151 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import List, Tuple
+
+import numpy as np
+
+
+@dataclass
+class Node:
+    """
+    Represents a node in the search tree for the Traveling Salesperson Problem.
+
+    Attributes
+    ----------
+    level : int
```

Comment:
There is no need to repeat the type annotations of the arguments in the docstring as well.

The Numpy documentation still hasn't been updated to accommodate for type hints, but there is [an open issue](https://github.com/numpy/numpydoc/issues/196) suggesting to do exactly this, with good approval as I can see. Also, they suggest to refer to [sphinx-autodoc-typehints](https://github.com/tox-dev/sphinx-autodoc-typehints), which also suggests to avoid type repetitions in the docstring.

So,
```suggestion
    level
```

and do same with the others.

Reply:
Thanks for picking this up! It has been fixed!